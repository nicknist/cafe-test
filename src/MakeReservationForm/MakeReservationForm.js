import React, { Component } from 'react';

class MakeReservationForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      date: '',
      time: '',
      guestNumber: ''
    }
  }

  changeInput = (key, value) => {
    this.setState({ [key]: value })
  }

  render() {
    return (
      <div className='res-form'>
        <input id='name' name='name' placeholder='Name' onChange={(e) => this.changeInput(e.target.name, e.target.value)}/>
        <input id='date' type='date' name='date' placeholder='date' onChange={(e) => this.changeInput(e.target.name, e.target.value)}/>
        <input id='time' type='time' name='time' placeholder='time' onChange={(e) => this.changeInput(e.target.name, e.target.value)} />
        <input id='guestNumber' name='guestNumber' type='number' placeholder='Number of Guests' onChange={(e) => this.changeInput(e.target.name, e.target.value)} />
        <button onClick={() => this.props.addReservation(this.state)}>Make Reservation</button>
      </div>
    )
  }
}

export default MakeReservationForm;
