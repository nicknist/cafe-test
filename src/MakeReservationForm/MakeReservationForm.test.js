import React from 'react';
import ReactDOM from 'react-dom';
import MakeReservationForm from './MakeReservationForm';
import { shallow } from 'enzyme';

describe('MakeReservationForm', () => {
  let wrapper;
  let mockAddReservation;

  beforeEach(() => {
    mockAddReservation = jest.fn();
    wrapper = shallow(<MakeReservationForm addReservation={mockAddReservation}/>);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should run addReservation with state when button is clicked', () => {
    wrapper.find('button').simulate('click');
    expect(mockAddReservation).toHaveBeenCalledWith(wrapper.state());
  })

  it('should update state when changeInpu is run', () => {
    expect(wrapper.state('name')).toEqual('');

    wrapper.instance().changeInput('name', 'Uncle Robbie')

    expect(wrapper.state('name')).toEqual('Uncle Robbie');
  })
})
