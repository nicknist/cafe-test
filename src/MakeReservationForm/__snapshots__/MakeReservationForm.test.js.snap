// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MakeReservationForm should match the snapshot 1`] = `
<div
  className="res-form"
>
  <input
    id="name"
    name="name"
    onChange={[Function]}
    placeholder="Name"
  />
  <input
    id="date"
    name="date"
    onChange={[Function]}
    placeholder="date"
    type="date"
  />
  <input
    id="time"
    name="time"
    onChange={[Function]}
    placeholder="time"
    type="time"
  />
  <input
    id="guestNumber"
    name="guestNumber"
    onChange={[Function]}
    placeholder="Number of Guests"
    type="number"
  />
  <button
    onClick={[Function]}
  >
    Make Reservation
  </button>
</div>
`;
